rollup.config.js

```diff
@@ -0,0 +1,127 @@
+const fs = require('fs');
+const {join} = require('path');
+
+// A Rollup plugin which locates modules using the Node resolution algorithm,
+// for using third party modules in node_modules
+const {nodeResolve} = require('@rollup/plugin-node-resolve');
+
+// A Rollup plugin to convert CommonJS modules to ES6, so they can be included
+// in a Rollup bundle
+const commonjs = require('@rollup/plugin-commonjs');
+
+// A Rollup plugin to minify generated ES bundles. Uses terser under the hood.
+const {terser} = require('rollup-plugin-terser');
+
+// A Rollup plugin to import CSS and inject it into the <head>
+const postcss = require('rollup-plugin-postcss');
+
+// A Rollup plugin to mark modules for Istanbul code coverage.
+const istanbul = require('rollup-plugin-istanbul');
+
+// A Rollup plugin which loads virtual modules from memory.
+const virtual = require('@rollup/plugin-virtual');
+
+/**
+ * Virtual imports made available to all bundles. Used for site config and globals.
+ */
+const site = require('./src/site/_data/site');
+const isProd = process.env.ELEVENTY_ENV === 'prod';
+const buildVirtualJSON = require('./src/build/virtual-json');
+const virtualImports = {
+  webdev_analytics: {
+    id: isProd ? site.analytics.ids.prod : site.analytics.ids.staging,
+    dimensions: site.analytics.dimensions,
+    version: site.analytics.version,
+  },
+  webdev_config: {
+    isProd,
+    env: process.env.ELEVENTY_ENV || 'dev',
+    version: 'v' + new Date().toISOString().replace(/[\D]/g, '').slice(0, 12),
+    firebaseConfig: isProd ? site.firebase.prod : site.firebase.staging,
+  },
+};
+
+// Each page type has an entrypoint js file which imports all of the custom
+// elements for that page. Most pages use the default.js entrypoint, but some
+// pages like /measure, /newsletter, etc. are special and require additional
+// elements.
+const pagesDir = './src/lib/pages/';
+const pages = fs.readdirSync(pagesDir, 'utf-8').map((p) => join(pagesDir, p));
+
+const devConfig = {
+  input: ['./src/lib/app.js', ...pages],
+  output: {
+    dir: 'dist/js',
+    format: 'esm',
+  },
+  watch: {
+    // By default rollup clears the console on every build. This disables that.
+    clearScreen: false,
+  },
+  plugins: [
+    virtual(buildVirtualJSON(virtualImports)),
+    nodeResolve(),
+    commonjs(),
+    postcss(),
+  ],
+};
+
+const productionConfig = {
```

Comment:
The configs are basically the same that I personally would build them programatically, e.g.:

```js
const config = {
  /* common stuff */
  plugins: [...];
};

// Build the config for our production or dev build.
if (isProduction) {
  config.plugins.push(terser());
} else {
  // etc
```

... but I'll confess it's a stylistic choice.